### DISCUSSION TITLE
Disable LEF Warning Messages

### DISCUSSION THREAD

**Discussion by abaababa** [The-OpenROAD-Project/OpenROAD#4289] [Q&A] [answered]
Hi all,

I am trying to load custom lef/lib files for pnr, and my lef files contain some properties unsupported by openroad. I got too much warning messages from this and the unsupported properties didn't have effect on my pnr results. Is there any way to disable the LEF warning messages temporarily? I checked the documents but failed to find an option that can turn off the warnings. Sorry if I missed it from the documents.

Thanks in advance!

**Comment by maliberty** [comment]
There is no method to do that currently.  No single message will print more than 1k times.

**Reply by abaababa** [reply]
I see, thanks for the information.

**Comment by luarss** [comment, marked as answer]
@maliberty  Correct me if I am wrong, but doesn't Tcl function `suppress_message` do this? 

Edit: i guess there's no function that achieves full suppression for all warnings, what OP requested

https://github.com/The-OpenROAD-Project/OpenROAD/blob/28892de3bef1af52f3371d547f752acf0af20021/test/helpers.tcl#L203

Corresponding source:
https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/utl/src/Logger.cpp

**Reply by maliberty** [reply]
Good point - you could use that for specific messages.  There is no flag to use it for all LEF parser messages though.
